src/components/Slider/services/getLaunches.ts

```diff
@@ -0,0 +1,37 @@
+import { fetchData } from '../../../shared/api/services/fetchData';
+import {
+  sliderGetPreviousLaunchesError,
+  sliderGetPreviousLaunchesSuccess,
+  sliderGetUpcomingLaunchesError,
+  sliderGetUpcomingLaunchesSuccess,
+  sliderLoadData,
+} from '../redux/sliderActionsCreators';
+import { Endpoints } from '../../../shared/api/constants/endpoints';
+import { launchesParams } from '../constants/constants';
+import type { Dispatch } from 'redux';
+import type { SliderAllActions } from '../types/SliderTypes';
+
+export const getLaunches =
+  (params: string) =>
+  async (dispatch: Dispatch<SliderAllActions>): Promise<void> => {
+    try {
+      dispatch(sliderLoadData());
+      if (params === launchesParams.UPCOMING) {
+        const upcomingLaunches = await fetchData(
+          Endpoints.GO_FOR_LAUNCHES_ENDPOINT,
+        );
+        dispatch(sliderGetUpcomingLaunchesSuccess(upcomingLaunches));
+      } else {
+        const upcomingLaunches = await fetchData(
+          Endpoints.PREVIOUS_LAUNCHES_ENDPOINT,
+        );
+        dispatch(sliderGetPreviousLaunchesSuccess(upcomingLaunches));
+      }
+    } catch (error) {
+      if (params === launchesParams.UPCOMING) {
+        dispatch(sliderGetUpcomingLaunchesError((error as Error).message));
+      } else {
+        dispatch(sliderGetPreviousLaunchesError((error as Error).message));
+      }
+    }
+  };
```

Comment:
You don't need params as an argument here and check if it is previous or upcoming. You have added complicated unnecessary logic. Just add two requests side by side. 
Actually, we don't need different actions for upcoming and previous launches, we need all data for the slider, don't we?

Reply:
discussed